java/src/jmri/util/swing/ValidatedTextField.java

```diff
@@ -309,19 +309,16 @@ private void exitFieldColorizer() {
 
             if ((getText() == null) || (getText().length() == 0)) {
                 // handle 0-length current value; 0-length is allowed
-                if (allow0Length == true) {
+                if (allow0Length) {
                     if ((lastQueryValue == null) || (lastQueryValue.length() == 0)) {
                         setBackground(COLOR_BG_UNEDITED);
-                        return;
                     } else {
                         setBackground(COLOR_BG_EDITED);
-                        return;
                     }
-                } else {
-                    // 0-length current value; 0-length is not allowed
-                    // (validator should prevent from getting to this case)
-                    return;
                 }
+                // 0-length current value; 0-length is not allowed
+                // (validator should prevent from getting to this case)
+                return;
             }
```

Comment:
I ask since I think this looks strange, but I might be wrong.

On line 312, you check if `allow0Length` is true. But on line 319, you have a comment `0-length is not allowed`.

If `allow0Length` is true, you still get to the line 319 that says that 0-length is not allowed.

Reply:
No problem. Code is not new/mine, but I applied the IDE hint to simplify the if, and extract common part (return). In the case you point out, nothing is actually done (empty else) so I took that out too.

Reply:
I suggest that you remove the two lines 319 - 320:
```
// 0-length current value; 0-length is not allowed
// (validator should prevent from getting to this case)
```
I think these adds confusion and no real value after your refactoring.